Adds genre to Image, Document, and StudentWork.

scherztc opened the pull request:
Fixes #1102 
Fixes #1273 

Summary:

Adds the genre as a controlled vocabulary to Images, Documents, and Student Works as it is in Scholar 2.x.  Removes the resource_type from all work types.

jamesvanmil commented:
Two notes - for Genre, we haven't used 'other' as an option.

For Student Works, we have had both a 'Work Type' field and a 'genre' field - did we decide to scrap one of these? I'm only seeing the 'genre' content.

scherztc commented:
@jamesvanmil :  I removed "Other" for Image and Document, but left it in for Student Work.  Since we got approval from POs to remove complex Student Work faceting, it was my understanding that we would only need the genre field for Student Works.  I kept the genre field and adjusted the label to use Work Types similar to labeling and help in Scholar 2.x.

jamesvanmil commented on app/helpers/sufia_helper.rb:
```diff
@@ -21,7 +21,7 @@ def sorted_college_list_for_etds
   def sorted_college_list_for_degrees
     COLLEGE_AND_DEPARTMENT["current_colleges_for_degrees"].keys.collect do |k|
       COLLEGE_AND_DEPARTMENT["current_colleges_for_degrees"][k]["label"]
-    end.sort << "Other"
```
We need to keep this "Other"

jamesvanmil commented on app/helpers/sufia_helper.rb:
```diff
@@ -37,7 +37,7 @@ def sorted_college_list_for_other_works
     )
     list.keys.collect do |k|
       list[k]["label"]
-    end.sort << "Other"
```
Need this one "Too"

jamesvanmil commented on app/views/curation_concerns/base/_attribute_rows.html.erb:
```diff
@@ -1,8 +1,8 @@
-<%= presenter.attribute_to_html(:department, render_as: :faceted) %>
-<%= presenter.attribute_to_html(:admin_set, render_as: :faceted) %>
+<%= presenter.attribute_to_html(:contributor) %>
+<%= presenter.attribute_to_html(:college) %>
```
I think we'd want to keep the faceted label on this and department - keep these as links from the show page?

jamesvanmil commented:
In Scholar 2, the "Other" label actually mapped to Generic Work on student work types - Other might not make sense in this new context.

Otherwise, I've added some line-wise notes, and the new helper `sorted_genre_list_for_works` needs some spec work.

scherztc commented:
@jamesvanmil : Ready for review.  I removed Other from Student Work and cleaned up the Help Text, made the line item changes, and wrote some test for the sufia_helper_spec.

jamesvanmil commented:
Hey sorry for the delay - one other note. The College lists generated by the helper need to hold on to "Other" - this is for folks who aren't in an academic unit.

newmanld commented:
In regard to genre field, See #1315 and #1314 which asked that we consult with metadata SME.  Did we do so, and did Metadata SME still want a controlled list, or free text with examples in help,  for genre on any or all work types (including ETDs)?  Maintain differences in a controlled list (the list has different choices by work type in Scholar 2.x)?  Is the label meant to be 'type' on one work type (ETD), 'Work type' on another and 'Genre on another'?  Please update metadata spreadsheet with all decisions as well, including values of a drop down list, labels and help text (and examples won't apply if it is changed to a controlled list instead of free text): https://docs.google.com/spreadsheets/d/1biO5sm4WVK9p01jCFDzNqTQ1mF_33MEHbpCjIVaOU3M/edit?usp=sharing.  Thanks.
